include/re_atomic.h

```diff
@@ -7,40 +7,346 @@
 #ifndef RE_H_ATOMIC__
 #define RE_H_ATOMIC__
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 /* C11 */
 #if defined(HAVE_ATOMIC) && !defined(__STDC_NO_ATOMICS__)
+
 #include <stdatomic.h>
 
-/* With MSVC volatile is atomic */
-#elif defined(_MSC_VER)
-#define RE_ATOMIC volatile
-#define atomic_load(object) InterlockedOr((object), 0)
+#define RE_ATOMIC _Atomic
+
+#define RE_ATOMIC_BOOL_LOCK_FREE ATOMIC_BOOL_LOCK_FREE
+#define RE_ATOMIC_CHAR_LOCK_FREE ATOMIC_CHAR_LOCK_FREE
+#define RE_ATOMIC_WCHAR_T_LOCK_FREE ATOMIC_WCHAR_T_LOCK_FREE
+#define RE_ATOMIC_SHORT_LOCK_FREE ATOMIC_SHORT_LOCK_FREE
+#define RE_ATOMIC_INT_LOCK_FREE ATOMIC_INT_LOCK_FREE
+#define RE_ATOMIC_LONG_LOCK_FREE ATOMIC_LONG_LOCK_FREE
+#define RE_ATOMIC_LLONG_LOCK_FREE ATOMIC_LLONG_LOCK_FREE
+#define RE_ATOMIC_POINTER_LOCK_FREE ATOMIC_POINTER_LOCK_FREE
+
+#define re_memory_order_relaxed memory_order_relaxed
```

Comment:
For C11 threads we decided to reuse the original function names. I think we should do the same for C11 atomic.

Reply:
I would like not to define macros and functions that clash with standard names. This is already a problem with `NULL`, `bool`, `true` and `false` that I mentioned before.

Besides, the macros are not implementable in a size-agnostic way on MSVC. Its intrinsics support only a fixed size of atomics, and there are different sets of atomics for different sizes and memory order types.